website/static/js/pages/project-dashboard-page.js

```diff
@@ -213,22 +238,26 @@ $(document).ready(function () {
         }
     });
 
-    //Clear input fields on Add Component Modal
-    $('#confirm').on('click', function () {
-        $('#alert').text('');
-        $('#title').val('');
-        $('#category').val('');
-    });
-
-    // only focus input field on modals when not IE
-    $('#newComponent').on('shown.bs.modal', function(){
-        if(!$osf.isIE()){
-            $('#title').focus();
-        }
-    });
+    loadCategories().then(function(categoryList) {
+        var addProjectTemplate = m.component(AddProject, {
+            buttonTemplate: m('.btn.btn-sm.btn-default[data-toggle="modal"][data-target="#addSubComponent"]', {onclick: function() {
+                $osf.trackClick('project-dashboard', 'add-component', 'open-add-project-modal');
+            }}, 'Add Component'),
+            modalID: 'addSubComponent',
+            title: 'Create new component',
+            parentID: window.contextVars.node.id,
+            parentTitle: window.contextVars.node.title,
+            categoryList: categoryList,
+            stayCallback: function () {
+                var ap = this; // AddProject controller
+            },
+            trackingCategory: 'project-dashboard',
+            trackingAction: 'add-component',
+            contributors: window.contextVars.node.contributors,
+            currentUserCanEdit: window.contextVars.currentUser.canEdit
+        });
 
-    $('#newComponent').on('hidden.bs.modal', function(){
-        $('#newComponent .modal-alert').text('');
+        m.mount(document.getElementById('newComponent'), m.component(addProjectTemplate, {wrapperSelector : '#addSubComponent'}));
```

Comment:
Yeah, I thought about that. Unfortunately, the categoryList is passed into the AddProjectPlugin which creates the button. I can play with it a little though and see if I can get the button to render when the page loads.